Mathlib/Combinatorics/SimpleGraph/VertexCover.lean

```diff
@@ -0,0 +1,192 @@
+/-
+Copyright (c) 2025 Vlad Tsyrklevich. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vlad Tsyrklevich
+-/
+import Mathlib.Combinatorics.SimpleGraph.Clique
+import Mathlib.Data.ENat.Lattice
+import Mathlib.Data.Set.Card
+
+/-!
+# Vertex cover
+
+A *vertex cover* of a simple graph is a set of vertices such that every edge is incident to at least
+one of the vertices in the set.
+
+## Main definitions
+
+* `SimpleGraph.IsVertexCover G C`: Predicate that `C` is a vertex cover of `G`.
+* `SimpleGraph.minVertexCover G`: The size of the minimal vertex cover for `G`.
+-/
+
+namespace SimpleGraph
+
+variable {V W : Type*} {G : SimpleGraph V} {H : SimpleGraph W}
+
+section IsVertexCover
+
+/-- `C` is a vertex cover of `G` if every edge in `G` is incident to at least one vertex in `C`. -/
+def IsVertexCover (G : SimpleGraph V) (c : Set V) : Prop :=
+  ∀ v w : V, G.Adj v w → v ∈ c ∨ w ∈ c
```

Comment:
Best practice is to use semi-implicit arguments in this situation. Compare eg `Function.Injective`
```suggestion
  ∀ ⦃v w⦄, G.Adj v w → v ∈ c ∨ w ∈ c
```